### DISCUSSION TITLE
App directory / React server component support

### DISCUSSION THREAD

**Discussion by b0o** [arlyon/stailwc#12] [Q&A] [answered]
Hey! This project looks exciting.

I was previously using twin.macro, but after upgrading to Next.js 13 and migrating to the app directory, I was disappointed that I couldn't use twin in server components (not necessarily twin's fault, neither emotion nor styled-components support server components yet). Since almost all of my pages use tailwind, this meant I didn't get to take advantage of server components at all.

I just finished gutting twin.macro from my project and using vanilla tailwind, which enabled me to use server components and transition from babel to swc but using tailwind via className leaves a lot to be desired.

I have yet to test out stailwc, but judging by the fact that you're using emotion, I'm guessing the story is similar to twin.macro. Do you have any particular plans to support server components in the app directory, or is this totally up to emotion adding support?

**Comment by arlyon** [comment, marked as answer]
This also relies on emotion for the machinery (as well as the swc emotion plugin). From a quick search, you may be interested in following this issue (https://github.com/vercel/next.js/issues/41994) for updates on when you can use stailwc for this purpose.
